pkgs/applications/misc/ilspy/default.nix

```diff
@@ -0,0 +1,99 @@
+{ lib
+, stdenv
+, pkgs
+
+, fetchzip
+, unzip
+, autoPatchelfHook
+, makeWrapper
+, makeDesktopItem
+
+#, stdenv.cc.cc.lib
+, lttng-ust
+, libkrb5
+, zlib
+, fontconfig
+
+, openssl_1_1
+, xorg
+, libX11
+, libICE
+, libSM
+, icu
+}:
+stdenv.mkDerivation rec {
+  pname = "ILSpy";
+  version = "7.2-rc";
+  description = ".NET assembly browser and decompiler.";
+
+  owner = "icsharpcode";
+  repo = "AvaloniaILSpy";
+
+  src = fetchzip {
+    url = "https://github.com/${owner}/${repo}/releases/download/v${version}/Linux.x64.Release.zip";
+    sha256 = "sha256:1crf0ng4l6x70wjlz3r6qw8l166gd52ys11j7ilb4nyy3mkjxk11";
+  };
+
+  nativeBuildInputs = [
+    unzip
+    autoPatchelfHook
+    makeWrapper
+  ];
+
+  buildInputs = [
+    stdenv.cc.cc.lib
+    lttng-ust
+    libkrb5
+    zlib
+    fontconfig
+  ];
+
+  libraryPath = lib.makeLibraryPath [
+    openssl_1_1
+    libX11
+    libICE
+    libSM
+    icu
+  ];
+
+  unpackPhase = ''unzip -qq ${src}/ILSpy-linux-x64-Release.zip'';
```

Comment:
you can just use `$src` variable within bash

also I'm fairly sure it'll just auto unzip when you have `unzip` in `nativeBuildInputs`

```suggestion
```

might need to do some sourceRoot stuff

Reply:
This was not my experience when constructing the package.

Reply:
How come the preference for using the bash variable over the nix one though?

Reply:
This should happen automatically. If not please try the suggested variables.

Reply:
Verified just for kicks. Default unpack phase hasn't started unzipping nested zips since the package was originally authored.